classes/User/User.mjs

```diff
@@ -22,51 +22,102 @@ export class User {
   async getUserById() {
     // returns user's public info
     // find() will be replaced with getById() when the find error in DatabaseController() is fixed
-    let user = await database.find({
+
+    
+    if (!this.id) {
+      err_out.message = "No user ID found. Instantiate User class with a proper ID as follows new User(UserId)"
+      err_out.status = 400
+      throw err_out
+    }
+
+    return  database.find({
       _id: this.id,
       "@type": "User"
     })
-    user = user[0]
-    this.id = user?._id
-    const publicUser = includeOnly(user, "profile", "_id")
-    this.user = publicUser
-    return publicUser
+    .then(resp => {
+      if (resp instanceof Error) { 
+        throw resp;
+      } 
+      const user = resp[0]
+      this.id = user?._id 
+      const publicUser = includeOnly(user, "profile", "_id")
+      this.user = publicUser
+      return publicUser 
+    })
+    .catch(err => { 
+      throw err; 
+    });
+ 
   }
 
   async getSelf() {
-    // returns full user object, only use this when the user is unauthenticated i.e, logged in and getting himself.
-
-    const user = await database.find({
+    // returns full user object, only use this when the user is unauthenticated i.e, logged in and getting himself. 
+    return  database.find({
       _id: this.id,
       "@type": "User"
     })
-    return user[0]
+    .then(resp => {
+      if (resp instanceof Error) { 
+        throw resp;
+      } 
+      return resp[0];
+    })
+    .catch(err => { 
+      throw err; 
+    });
+ 
   }
 
   async updateRecord(data) {
     // updates user object. use with PUT or PATCH on authenticated route only
-    if (!data) return
-    const previousUser = this.user
-    const newRecord = {...previousUser, ...data}
-    const updatedUser = await database.update(newRecord)
-    return updatedUser
-  }
+    if (!data) {
+      err_out.message = "No payload provided"
+      err_out.status = 400 
+      throw err_out
+    }
+
+    const previousUser = await this.getSelf() 
+    const newRecord = {...previousUser, ...data} 
+
+    return database.update(newRecord).then((resp)=>{
+      console.log("entered here")
+      if(resp instanceof Error){ 
+        throw(resp)
+      }
+     return resp
+      
+    }).catch((error)=> Promise.reject(error))
```

Comment:
You do this different than the other ones.  In the other ones you do 

```
.catch(err => { 
     throw err
})
```

Unless there is a logical reason to do it this way, stay consistent.